util/design/lib/OtpMemImg.py

```diff
@@ -413,6 +421,52 @@ def streamout_partition(self, part):
         # in the memory hex file.
         return data, annotation
 
+    def validate_data_perm(self, data_perm):
+        '''Validate data permutation option'''
+
+        # Byte aligned total width after adding ECC bits
+        total_bytelen = (
+            (self.lc_state.config['secded']['data_width'] +
+             self.lc_state.config['secded']['ecc_width'] + 7) // 8)
+        total_bitlen = total_bytelen * 8
```

Comment:
I'd consider extracting some locals, which would make this a bit less crazy looking:
```suggestion
        secded_cfg = self.lc_state.config['secded']
        raw_bitlen = secded_cfg['data_width'] + secded_cfg['ecc_width']
        total_bitlen = ((raw_bitlen + 7) // 8) * 8
```

Reply:
Thanks - done!